src/infill.cpp

```diff
@@ -402,10 +455,32 @@ void Infill::generateCrossInfill(const SierpinskiFillProvider& cross_fill_provid
     }
 }
 
+void Infill::generateCrossInfill(const SierpinskiFillProvider& cross_fill_provider, Polygons& result_lines)
+{
+    Polygons result_polygons;
+    if (mirror_offset)
+    {
+        const coord_t offset = infill_multiplier % 2 == 0 ? outline_offset : outline_offset + infill_line_width / 2 ;
+        result_polygons.add(outer_contour.offset(offset));
+        generateCrossInfill(cross_fill_provider, result_polygons, result_lines);
+        for (PolygonRef poly_line : result_polygons)
+        {
+            for (unsigned int point_idx = 1; point_idx < poly_line.size(); point_idx++)
```

Comment:
Use `size_t` for comparison with `size()`.

Reply:
Now you're getting the hang of the 'maximalist' approach, since I didn't touch this. But I agree with your assessment.

This will however result at least two extra `static_cast`s since the bracket operator of PolygonRef takes an `unsigned int` Which is wrong imo. But changing that to `size_t` will be a huge change in the code. The compiler will trip over al ot of ambiguous warnings
`error: use of overloaded operator '[]' is ambiguous (with operand types 'cura::PolygonRef' and 'unsigned int')` because we do it wrong in a lot of places. This could also result in a bigger footprint since unsigned int is 32bit and size_t is 64bit

I will compare against `size_t` and static_cast to `unsigned int`